bvm/ballerina-logging/src/main/java/org/ballerinalang/logging/BLogManager.java

```diff
@@ -65,8 +65,10 @@
     public static final String BALLERINA_ROOT_LOGGER_NAME = "ballerina";
     public static final int LOGGER_PREFIX_LENGTH = BALLERINA_ROOT_LOGGER_NAME.length() + 1; // +1 to account for the .
     private static final Pattern varPattern = Pattern.compile("\\$\\{([^}]*)}");
+    private static boolean strandLogLevelPropagate;
 
     private Map<String, BLogLevel> loggerLevels = new HashMap<>();
+    private Map<Integer, BLogLevel> strandLoggerLevels = new HashMap<>();
```

Comment:
I think we need to hold this config information inside the strand. If not this map with grow as we are not removing the entry once the strand is dead.

Reply:
Updated in 3110c83